drivers/auxdisplay: add support for LCD1602

leonmariotto opened the pull request:
Move backlight i2c controller address into DTS configuration to be able to use jhd1313 driver for dfrobot's LCD1602.

xingrz reviewed (changes requested):
Please avoid reformatting the entire file unnecessarily. If reformatting is truly needed, it should be done in a separate commit.

Commit:
drivers/auxdisplay: add support for dfrobot LCD1602 I2C module

Move backlight i2c controller address into DTS configuration
to be able to use jhd1313 driver for dfrobot's LCD1602.

Signed-off-by: Leon Mariotto <leon2mariotto@gmail.com>

Commit:
drivers/auxdisplay: format jhd1313 driver.

Format jhd1313 driver to comply with CI rules.

Signed-off-by: Leon Mariotto <leon2mariotto@gmail.com>

leonmariotto commented:
For additional information,
I've tested it on a nucleo_h563zi :
![IMG_20250508_142825](https://github.com/user-attachments/assets/298f0a7f-d8a4-471a-9712-ada097f5b13d)
The 2 screen (jhd1313 and dfrobot's LCD1602 RGB) are almost identical, only i2c addresses changes.
DFRobot's LCD1602 RGB datasheet : https://raw.githubusercontent.com/DFRobot/DFRobot_RGBLCD/master/DFR0464%20Datasheet.pdf
JHD1313 datasheet :  https://files.seeedstudio.com/wiki/Grove_LCD_RGB_Backlight/res/JHD1313%20FP-RGB-1%201.4.pdf
